wormhole-connect/src/views/v2/Bridge/AmountInput/index.tsx

```diff
@@ -183,13 +204,66 @@ const AmountInput = (props: Props) => {
         )}
       </Stack>
     );
-  }, [isInputDisabled, props.tokenBalance, sendingWallet.address]);
+  }, [
+    classes.balance,
+    isInputDisabled,
+    props.isFetchingTokenBalance,
+    props.tokenBalance,
+    sendingWallet.address,
+  ]);
 
   const handleChange = useCallback((newValue: string): void => {
-    dispatch(setAmount(newValue));
     setAmountInput(newValue);
   }, []);
 
+  const tokenPriceAdornment = useMemo(() => {
+    const price = calculateUSDPrice(
+      getTokenPrice,
+      Number(amountInput),
```

Comment:
we have special handling for when input starts with `.`, so users can add a fractional amount quickly. this code doesn't use that handling and throws NaN into the UI. see `onInnerChange`.

<img width="163" alt="image" src="https://github.com/user-attachments/assets/eb409be1-810c-414f-b02d-09ca83939355" />